pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/StudyBuilderCli.java

```diff
@@ -212,6 +228,25 @@ private void run(String[] args) throws Exception {
             }
             runEmails(cmd, cfgPath, studyCfg, varsCfg);
             return;
+        } else if (cmd.hasOption(OPT_SYNC_CLIENT_SECRETS)) {
+            if (isDryRun) {
+                throw new DDPException("-" + OPT_SYNC_CLIENT_SECRETS + " does not support -" + OPT_DRY_RUN);
+            }
+
+            final var rotateSecrets = cmd.hasOption(OPT_ROTATE_SECRETS);
+
+            if (rotateSecrets) {
+                final var shouldContinue = promptForConfirmation("A client secret rotation can not be undone. Do you wish to proceed?");
```

Comment:
I was going to suggest this confirmation check because its too powerful like invalidate study.
Glad you added it